### PR TITLE
chore(deps): bump react-native 0.70.13

### PR DESCRIPTION
**Pull request by araujobarret**
### Description

This PR bumps react-native to 0.70.13. One blocker I've found for Android was the outdated tipsi-stripe library, ~I replaced it with the official library making the adjustments and adjusting the tests~ the solution was to add `jetifier` to the install script as we do in the android build script.
This PR is divided into 3 commits:
- [1st](https://github.com/artsy/eigen/commit/eac66f36701343c87bc20a8121398c6b55ca87c0): rn-diff from [upgrade-helper](https://react-native-community.github.io/upgrade-helper/?from=0.69.12&to=0.70.13)
- [2nd](https://github.com/artsy/eigen/commit/7786c08ca2466e1819183382f193b9c5240babef): ~`tipsi-stripe` replacement with `@stripe/react-native`~ reverted on https://github.com/artsy/eigen/pull/9189/commits/50fba80768d36f4dbead52f8d18f63751f1dfc0e
- [3rd](https://github.com/artsy/eigen/commit/a24e373e279903ea98990d4ecea1b5d1392f1a0b): Fix missing values used in XML and bump kotlin version to `1.8.0`.
- [4th](https://github.com/artsy/eigen/pull/9189/commits/efb0dc3af028e9ecc1ae23b58be0805e143e9a0c): replaces `Dimensions.removeEventListener` with the new format, this was generating some errors and breaking the tests

### PR Checklist

- [x] I have tested my changes on **iOS** and **Android**.
- [x] I hid my changes behind a **[feature flag]**, or they don't need one.
- [x] I have included **screenshots** or **videos**, or I have not changed the UI.
- [x] I have added **tests**, or my changes don't require any.
- [x] I added an **[app state migration]**, or my changes do not require one.
- [x] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [x] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [x] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

- bump react-native to 0.70.13 - araujobarret

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
### This PR contains the following changes:

- Cross-platform user-facing changes (bump react-native to 0.70.13 - araujobarret)
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against a24e373e279903ea98990d4ecea1b5d1392f1a0b
</p>

**Review by MounirDhahri — DISMISSED** [review]
Niiiiiiice 🔥

**Review by gkartalis — COMMENTED** [review]
woohoo, excited to test it out

**Comment by araujobarret** [comment]
Android release builds are broken in Android because of the Sentry: https://github.com/getsentry/sentry-java/issues/2601
We need to upgrade Sentry to enable this PR 🧽.

**Comment by gkartalis** [comment]
Created a ticket for upgrading sentry here -> https://artsyproduct.atlassian.net/browse/PHIRE-207

File: src/app/Components/Bidding/Screens/ConfirmBid/index.tsx [hunk 1]
```diff
@@ -183,26 +185,22 @@ export class ConfirmBid extends React.Component<ConfirmBidProps, ConfirmBidState
...
+      type: "Card",
+      name: billingAddress?.fullName,
+      address: {
+        line1: billingAddress?.addressLine1,
+        line2: billingAddress?.addressLine2,
+        city: billingAddress?.city,
+        state: billingAddress?.state,
+        postalCode: billingAddress?.postalCode,
+        country: billingAddress?.country.shortName,
+      },
```
Comment (brainbicycle) [inline src/app/Components/Bidding/Screens/ConfirmBid/index.tsx:199]: 🤩

**Review by brainbicycle — COMMENTED** [review]
So a major react native upgrade, a major sentry upgrade and you got us off a forked dependency? Legend 💪 

Awesome work. 

I am wondering if it might make sense to separate out the stripe change into a separate pr similar to the sentry change just to keep the separation clean in case something goes wrong. 

Thanks for doing this!

File: src/app/Components/Bidding/Screens/CreditCardForm.tests.tsx [hunk 2]
```diff
@@ -89,16 +86,14 @@ it("is enabled while the form is valid", () => {
   expect(component.findByType(Button).props.disabled).toEqual(false)
 })
 
-it("shows an error when stripe's API returns an error", () => {
+// Running this test isolated it's fine, but running altogether fails
+// needs to be re-visited to be fixed
+it.skip("shows an error when stripe's API returns an error", async () => {
```
[the remark this answers is not in the document]
Comment (brainbicycle) [reply src/app/Components/Bidding/Screens/CreditCardForm.tests.tsx:91]: got it thanks

**Comment by araujobarret** [comment]
I wondered that about Stripe too, but tipisi-stripe was breaking Android with some weird Java stuff, then the upgrade seemed like a quick win given no breaking changes. 
@gkartalis and I will be doing some QA once the betas are ready in the offer/buy/auction flow to make sure everything is in order, feel free to try out the betas once they're ready too 👍

**Comment by brainbicycle** [comment]
> I wondered that about Stripe too, but tipisi-stripe was breaking Android(during build time) with some weird Java stuff, then the upgrade seemed like a quick win given no breaking changes.
> @gkartalis and I will be doing some QA once the betas are ready in the offer/buy/auction flow to make sure everything is in order, feel free to try out the betas once they're ready too 👍

yeah I hear that not saying not to do the stripe change, just if possible put it into a separate PR and merge first, that way if there is breakage with credit cards or something down the line it is easy to isolate

**Comment by araujobarret** [comment]
Actually, I can do it that way, not sure if kotlin bump would work with the old React native version but I can try separating it for a revert in case of issues.

**Comment by araujobarret** [comment]
@brainbicycle I was working on retiring tipsi in another branch but found a blocker when replacing it with official stripe library(https://github.com/stripe/stripe-react-native/pull/1044).

The solution was to rever tipisi commit here and add jetifier to the install script to fix android -> androidx packaging translation from tipsi, with that the builds are 100% fine.

Going to generate the betas to test one more time and if everything is fine merge this PR.

**Pull request merged**